pkg/generators/servers.go

```diff
@@ -330,19 +331,65 @@ func (g *ServersGenerator) generateServerAdapterSource(resource *concepts.Resour
 			{{ $requestBodyParameters := requestBodyParameters . }}
 			{{ $requestBodyLen := len $requestBodyParameters }}
 			{{ $responseParameters := responseParameters . }}
+			{{ $requestQueryParameters := requestQueryParameters . }}
 	
 			func (a *{{ $adapterName }}) read{{ $requestName }}(r *http.Request) (*{{ $requestName }}, error) {
+				var err error
 				result := new({{ $requestName }})
-				result.query = r.Form
+				result.query = r.URL.Query()
 				result.path = r.URL.Path
 
+				{{ if $requestQueryParameters }}
+					{{ range  $requestQueryParameters }}
+						{{ $fieldName := fieldName . }}
```

Comment:
Instead of having this long list of "ifs" I think it would be better to have a `readerName` template function that receives the type of the parameter and returns `ReadInteger`, `ReadFloat`, etc. Then you can use the same coe for all parameters:

```
{{ fieldName := fieldName . }}
{{ readerName := readerName . }}
result.{{ $fieldName }}, err := {{ $readerName }}(query, "{{ $fieldName }}")
if err != nil {
        ...
}
```

Reply:
Also note that using `fieldName` to calculate the name of the query parameter isn't ideal. It is intended to generate Go names. If you have a query parameter named `type` it will generate `type_` to avoid a syntax error. And that would not work as expected. Consider using a different function to calculate these names.